Tests/YamsTests/EncoderTests.swift

```diff
@@ -322,6 +322,19 @@ class EncoderTests: XCTestCase { // swiftlint:disable:this type_body_length
         XCTAssertNil(t.n5)
     }
 
+    func testEncodingDateWithNanosecondGreaterThan999499977() throws {
+    #if _runtime(_ObjC) || swift(>=5.0)
+        let gregorian = Calendar(identifier: .gregorian)
+        let utc = TimeZone(identifier: "UTC")!
+        var dateComponents = gregorian.dateComponents(in: utc, from: Date())
```

Comment:
According to the condition, #192 occurs only 5 times in 10000 times. Nevertheless, I was able to find #192 because I saw that `testEncodingDate()` using `Date()` actually failed. This issue only involves nanoseconds. So I want to use random data as well as `testEncodingDate()` except `nanosecond`.

Reply:
Nice catch!